File: aspnet-in-practice/14/CS/Cache/Cache/Cache/CacheFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Reflection;
using System.Configuration;
using System.Runtime.Caching;
using Microsoft.Practices.Unity.Configuration;
using Microsoft.Practices.Unity;

namespace Cache
{
	public static class CacheFactory
	{
		private static IUnityContainer container;

		static CacheFactory()
		{
			container = new UnityContainer();
			var config = (UnityConfigurationSection)ConfigurationManager.GetSection("unity");
			config.Containers.Default.Configure(container);
		}

		public static ObjectCache Instance
		{
			get
			{
				return container.Resolve<ICacheBuilder>().GetInstance();
			}
		}
	}
}

File: aspnet-in-practice/14/VB/Cache/Cache/CacheFactory.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Reflection
Imports System.Configuration
Imports System.Runtime.Caching
Imports Microsoft.Practices.Unity.Configuration
Imports Microsoft.Practices.Unity

Public NotInheritable Class CacheFactory
	Private Sub New()
	End Sub
	Private Shared container As IUnityContainer

	Shared Sub New()
		container = New UnityContainer()
		Dim config = DirectCast(ConfigurationManager.GetSection("unity"), UnityConfigurationSection)
		config.Containers.Default.Configure(container)
	End Sub

	Public Shared ReadOnly Property Instance() As ObjectCache
		Get
			Return container.Resolve(Of ICacheBuilder)().GetInstance()
		End Get
	End Property
End Class